source/addonHandler/__init__.py

```diff
@@ -388,7 +389,7 @@ def enable(self, shouldEnable):
 			if self.name in state["pendingEnableSet"]:
 				# Undoing a pending enable.
 				state["pendingEnableSet"].discard(self.name)
-			# No need to disable an addon that is already disabled.
+			# No need to disable an addon that is already disabled, except if all add-ons are disabled.

```

Comment:
This comment is not necessary now.
```suggestion
			# No need to disable an addon that is already disabled.
```